### PR TITLE
[WIP] VIZ/EDU: how to analyze the forward model with PCA

### PR DESCRIPTION
**Pull request by dengemann**
Here I have some small example to investigate the effective dimensions in the lead field.
The script with its comments should be self-explanatory.
But, overall, I am a bit surprised about the relatively small number of dimensions.
Any feedback for improving the computation approach and correctness would be highly welcome.

cc @larsoner @agramfort @SherazKhan etc.

Here is the output:

screenshot 2018-11-21 13 10 19

### CODE REVIEW

File: examples/forward/plot_forward_rank.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+for picks, ch_type, color in zip(
+        (picks_grad, picks_mag, picks_eeg),
+        ('grad', 'mag', 'eeg'),
+        ('c', 'b', 'k')):
+
+    # use same number of channels while keeping spatial info.
+    exp_var_ratio = np.array([
+        _get_var_exp(
+            leadfield[np.random.choice(picks, size=n_chans, replace=False)])
+        for _ in range(n_draws)])
```
Comment (agramfort) [inline examples/forward/plot_forward_rank.py]: I don't get why you need to subsample

**Comment by agramfort** [comment]
yes but why not keeping all channels? if you add redundant channels it
won't increase the rank

**Comment by dengemann** [comment]
So you’re saying that resulting differences in the svd spectra not simply
reflect higher resolution through added sensors but true intrinsic
resolution?
The 99% cumulative variance explained dimensions are about 50, 25 and 18
for grad, mag, eeg if you take all sensors within each sensor type.

**Comment by agramfort** [comment]
yes

**Comment by dengemann** [comment]
@agramfort I removed subsampling and updated the comments a bit. I'd eventually like to add some more plots for exploring topographic patterns and visiting the effective rank by source label, but this can also be another PR.

screenshot 2018-11-26 00 34 16

**Comment by agramfort** [comment]
maybe @LauriParkkonen or @mshamalainen have an opinion or a ref for the best way to display this.  I am sure this has been done before.

**Comment by larsoner** [comment]
@dengemann are you up for polishing this for MRG? If not I'd say let's close and someone can resurrect if they are interested in the issue

**Pull request closed without merging**